classlib/src/main/java/org/teavm/classlib/java/util/TSpliterators.java

```diff
@@ -0,0 +1,77 @@
+/*
+ *  Copyright 2017 Alexey Andreev.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.teavm.classlib.java.util;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.function.IntConsumer;
+
+public class TSpliterators {
+
+    private TSpliterators() {
+    }
+
+    public static <T> TSpliterator<T> spliterator(Object[] array, int additionalCharacteristics) {
```

Comment:
Why didn't you implement Spliterator for this case?

Reply:
Not sure I understand. The "normal" Spliterators also doesn't implement Spliterator. Could it be that you misread TSpliterator instead of TSpliterators? If not, I'm probably misunderstanting something.

Reply:
You create extra list here, while you could just implement Spliterator interface that exposes array content directly. This would (1) consume less memory (2) reflect changes to underlying array. For the pt.2 you should write tests to ensure that your implementation works as in OpenJDK

Reply:
Changed implementation and added test for the behavior suggested in item 2.